security/login_link.rst

```diff
@@ -0,0 +1,510 @@
+.. index::
+   single: Security; Login link
+   single: Security; Magic link login
+
+How to use Passwordless Login Link Authentication
+=================================================
+
+Login links, also called "magic links", allow users to login without
+passwords. Whenever a user wants to login, a login link is generated and
+send to the user (e.g. using an e-mail). The user is logged in as soon as
+they visit that link.
+
+Login links are supported by Symfony when using the experimental
+authenticator system. You must
+:ref:`enable the authenticator system <security-enable-authenticator-manager>`
+in your configuration to use this feature.
+
+Using the Login Link Authenticator
+----------------------------------
+
+This guide assumes you have set-up security and have created a user object
+in your application. Follow :doc:`the main security guide </security>` if
+this is not yet the case.
+
+1) Configure the Login Link Authenticator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The login link authenticator is configured using the ``login_link`` option
+under the firewall. In the most minimal form, you have to configure a
+``check_route``:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/security.yaml
+        security:
+            firewalls:
+                main:
+                    login_link:
+                        check_route: login_check
+
+    .. code-block:: xml
+
+        <!-- config/packages/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+
+            <config>
+                <firewall name="main">
+                    <login-link check-route="login_check"/>
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                'main' => [
+                    'login_link' => [
+                        'check_route' => 'login_check',
+                    ],
+                ],
+            ],
+        ]);
+
+The ``check_route`` must be an existing route and is used by the
+authenticator to check the login link and authenticate the user.
```

Comment:
Maybe say something here about this will be the route that the magic link points to?

```rst
The ``check_route`` must be an existing route and it will be used to generate
the login link that will authenticate the user. You don't need a controller (or it
can be empty) because the login link authenticator will intercept requests to
this route:
```